extmod/network_lwip.c

```diff
@@ -90,6 +100,241 @@ mp_obj_t mod_network_nic_ifconfig(struct netif *netif, size_t n_args, const mp_o
+            if (MP_MAP_SLOT_IS_FILLED(kwargs, i)) {
+                mp_map_elem_t *e = &kwargs->table[i];
+                switch (mp_obj_str_get_qstr(e->key)) {
+                    case MP_QSTR_dhcp4: {
+                        if (e->value == mp_const_true) {
+                            if (dhcp_supplied_address(netif)) {
+                                dhcp_renew(netif);
+                            } else {
+                                dhcp_release_and_stop(netif);
+                                dhcp_start(netif);
+                            }
+                            uint32_t start = mp_hal_ticks_ms();
+                            while (!dhcp_supplied_address(netif)) {
+                                if (mp_hal_ticks_ms() - start > 10000) {
+                                    mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("timeout waiting for DHCP to get IP address"));
+                                }
+                                mp_hal_delay_ms(100);
+                            }
+                        } else {
+                            dhcp_release_and_stop(netif);
+                        }
+                        break;
+                    }
+                    #if LWIP_IPV6_DHCP6
+                    case MP_QSTR_dhcp6: {
+                        dhcp6_disable(netif);
+                        dhcp6_enable_stateless(netif);
+                        break;
+                    }
+                    #endif
+                    #if LWIP_IPV6_AUTOCONFIG
+                    case MP_QSTR_autoconf6: {
+                        netif_set_ip6_autoconfig_enabled(netif, e->value == mp_const_true);
+                        if (e->value == mp_const_true) {
+                            nd6_restart_netif(netif);
+                            uint32_t start = mp_hal_ticks_ms();
+                            while (1) {
+                                int found = 0;
+                                for (i = 1; i < LWIP_IPV6_NUM_ADDRESSES; i++) {
+                                    if (ip6_addr_isvalid(netif_ip6_addr_state(netif, i)) &&
+                                        !netif_ip6_addr_isstatic(netif, i)) {
+                                        found = 1;
+                                        break;
+                                    }
+                                }
+                                if (found) {
+                                    break;
+                                }
+                                if (mp_hal_ticks_ms() - start > 10000) {
+                                    mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("timeout waiting for DHCP to get IP address"));
+                                }
+                                mp_hal_delay_ms(100);
+                            }
+                        } else {
+                            // Skip link-local address in slot 0
+                            for (i = 1; i < LWIP_IPV6_NUM_ADDRESSES; i++) {
```

Comment:
Similarly comment to explain "Clear out any non-static addresses, skipping the link-local address in slot 0".

(That said, what if slaac is disabled, does it still reserve slot 0?)